### PR TITLE
Remove 'Redis' in error replies

### LINKED ISSUE #204: Occurrences of "Redis" in error replies
The following error messages are considered high risk breaking change, so they will be controlled by special compatibility configuration, thus they are **not in scope** in this issue:

    "-MISCONF Redis is configured to save RDB snapshots (...)"
    "-LOADING Redis is loading the dataset in memory"
    "-BUSY Redis is busy running a script (...)"

**All other error replies** containing "Redis" can be changed, either by removing "Redis" or changing it to "Valkey", "the server" or something else, depending on context.

### PR DESCRIPTION
**Pull request by zuiderkwast**
Low-risk error replies containing "Redis" are changed, **excluding the following ones that are instead covered in #274** (since they're high risk of causing problems):

* (not in scope) "-MISCONF Redis is configured to save RDB snapshots (...)"
* (not in scope) "-LOADING Redis is loading the dataset in memory"
* (not in scope) "-BUSY Redis is busy running a script (...)"

Additionally, error replies from `redis.call` in a Lua script are affected, such as

* "Please specify at least one argument for this redis lib call"
* "Wrong number of args calling Redis command from script"
* "Unknown Redis command called from script"
* "Invalid command passed to redis.acl_check_cmd()"

The name Redis is simply removed from these error message. In the last one above, "redis.acl_check_cmd()" is replaced by "server.acl_check_cmd()" in the error message.

Fixes #204

For release notes:

> The word "Redis" is removed in the following error replies:
> * "This Redis instance is not configured to use an ACL file. You may want to specify users via the ACL SETUSER command and then issue a CONFIG REWRITE (assuming you have a Redis configuration file set) in order to store users in the Redis configuration."
> * "Active defragmentation cannot be enabled: it requires a Redis server compiled with a modified Jemalloc like the one shipped by default with the Redis source requires a server compiled with a modified Jemalloc like the one shipped by default with the Valkey source distribution"
> * "Wrong number of args calling Redis command from script"
> * "Unknown Redis command called from script"
>
> The words "used by Redis" are removed in the error message
> * "The event loop API used by Redis is not able to handle the specified number of clients"
>
> The words "since Redis 4" are removed in the error message
> * "WAIT cannot be used with replica instances. Please also note that since Redis 4.0 if a replica is configured to be writable (which is not the default) writes to replicas are just local and are not propagated."
>
> The words "redis lib" are removed from in the error message
> * "Please specify at least one argument for this redis lib call"
>
> "Lua redis lib command" replaced with just "Command" in the error message
> * "Lua redis lib command arguments must be strings or integers"
>
> "redis.acl_check_cmd" replaced by "server.acl_check_cmd" in the following error message:
> * "Invalid command passed to redis.acl_check_cmd()"

### CODE REVIEW

**Comment by zuiderkwast** [comment]
@0del do you want to review this one?

File: tests/unit/acl.tcl
Comment (9bany) [inline tests/unit/acl.tcl:908]: i'm not sure
just suggest - uppercase for the first character: rename `instance` -> `Instance`

**Review by 9bany — APPROVED** [review]
LGTM !

File: tests/unit/acl.tcl
Comment (zuiderkwast) [reply tests/unit/acl.tcl:908]: This is a pattern in the test case. It's matching the error message "This instance is not configured to use an ACL file...." created in `acl.c`.

`*` is a wildcard character here.

Comment (9bany) [reply tests/unit/acl.tcl:908]: Thank you for your explanation ! Great 👍🏼

**Review by hwware — APPROVED** [review]

File: src/server.c
Comment (madolson) [inline src/server.c]: This is duplicated here: https://github.com/valkey-io/valkey/pull/205. 

I think this will be hardcoded a bunch of places. I don't want to break compatibility without thinking this through more.

Comment (madolson) [reply src/server.c]: I'm also concerned about LOADING error message. I would rather separate those out into their own review and mark it as a major decision.

Comment (zuiderkwast) [reply src/server.c]: How come?

If -LOADING breaks something, so can the other errors? Maybe all of it should be a major decision?

Clients are supposed to only look at the first uppercased word in the error. (I know not everyone does what they're supposed to do.)

Comment (zuiderkwast) [reply src/server.c]: @madolson I see [netty](https://github.com/netty/netty/blob/fee3c39d0835ec18d7aace8776db5c5952ce8272/codec-redis/src/main/java/io/netty/handler/codec/redis/FixedRedisMessagePool.java#L44) has some enum values with BUSY, LOADING and MISCONF containing "Redis".

What impact will it have to the client if we change this? It will still probably understand that it's an error, but will it act differently?

Comment (zuiderkwast) [reply src/server.c]: To me it looks like a string pool, i.e. an optimization. It will accept other messages too but with an extra allocation or so.

This is where it seems to be used: https://github.com/netty/netty/blob/4.1/codec-redis/src/main/java/io/netty/handler/codec/redis/RedisDecoder.java#L264 (read together with the other code linked above)

Comment (madolson) [reply src/server.c]: I'm saying people are actually looking for the whole string. I'm not confident that people are just looking for these error messages. I would rather split these out and review them independent, I think these are much more likely to be breaking changes.

Comment (zuiderkwast) [reply src/server.c]: Yeah they are breaking changes. We shall have them in the next major version IMO.

**Comment by PingXie** [comment]
I don't think we should make this change. We should use the indirection mechanism established in #47 instead. As we discussed in other threads, I am proposing to have one uber compilation flag such as '-DREDIS_COMAPT=1' to control the presence/absence of all breaking changes.

**Comment by zuiderkwast** [comment]
@valkey-io/core-team After decision made about #274, I've reduced the scope of this PR to only include "low risk" error replies. The "high risk" ones are handled in #274 instead.

I consider the scope of this PR (and the linked issue) already decided. It just needs a second review before merge.

**Review by enjoy-binbin — APPROVED** [review]

**Pull request merged**
